lib/components/paginated_item_carousel.dart

```diff
@@ -51,13 +51,13 @@ class _FutureItemCarouselState extends ConsumerState<PaginatedItemCarousel> {
     final isLastPage = useState(false);
 
     if (data.isEmpty && !isLastPage.value) {
-      isLoading.value = true;
+      if (isLoading.hasListeners) {
+        isLoading.value = true;
+      }
       widget.future(startIndex, widget.pageSize).then((value) {
         data.addAll(value);
         startIndex += widget.pageSize;
-        isLoading.value = false;
-
-        if (value.length < widget.pageSize) {
+        if (value.length < widget.pageSize && isLastPage.hasListeners) {
```

Comment:
.hasListeners should be used with "normal" ChangeNotifiers not with a Hook.

Reply:
no clue what this means, sorry. do you mind pushing the relevant change yourself so I know what you're talking about?

Reply:
calling a hook like isLoading.hasListeners produces a warning: The member 'hasListeners' can only be used within instance members of subclasses of 'package:flutter/src/foundation/change_notifier.dart'.

But you could use context.mounted instead. This checks if the current widget is mounted, which does essentially the same without warnings.